resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -1513,7 +1520,9 @@ private[spark] object Client extends Logging {
     }
 
     sys.env.get(ENV_DIST_CLASSPATH).foreach { cp =>
-      addClasspathEntry(getClusterPath(sparkConf, cp), env)
+      val newCp = cp.split(CLASS_PATH_SEPARATOR)
+        .filterNot(cpSet.contains).mkString(CLASS_PATH_SEPARATOR)
+      addClasspathEntry(getClusterPath(sparkConf, newCp), env)
```

Comment:
Try to deduplicate jars that have been added to `CLASSPATH` through `extraClassPath`

Reply:
Is it necessary to change it to 

```
      val newCp = if (Utils.isTesting) {
        cp.split(File.pathSeparator)
          .filterNot(cpSet.contains).mkString(File.pathSeparator)
      } else cp
```